content/docs/instrumenting/writing_exporters.md

```diff
@@ -316,83 +244,64 @@ Rather create new metrics each time. In Go this is done with
-sitting right beside it on the same machine. That means for every haproxy you
-run, you run a `haproxy_exporter` process. For every machine with a mesos
-slave, you run the mesos exporter on it (and another one for the master if a
-machine has both).
+sitting right beside it on the same machine. That means for every HAProxy you
+run, you run a `haproxy_exporter` process. For every machine with a Mesos
+worker, you run the [Mesos exporter](https://github.com/mesosphere/mesos_exporter) on it, and another one for the master, if a machine has both.
 
 The theory behind this is that for direct instrumentation this is what you’d be
 doing, and we’re trying to get as close to that as we can in other layouts.
 This means that all service discovery is done in Prometheus, not in exporters.
 This also has the benefit that Prometheus has the target information it needs
-to allow users probe your service with the blackbox exporter.
+to allow users probe your service with the [blackbox exporter](https://github.com/prometheus/blackbox_exporter).
 
 There are two exceptions:
 
 The first is where running beside the application your monitoring is completely
-nonsensical. SNMP, blackbox and IPMI are the main examples of this. IPMI and
-SNMP as the devices are effectively black boxes that it’s impossible to run
-code on (though if you could run a node exporter on them instead that’d be
-better), and blackbox as if you’re monitoring something like a DNS name there’s
-nothing to run on. In this case Prometheus should still do service discovery,
-and pass on the target to be scraped. See the blackbox and SNMP exporters for
-examples.
+nonsensical. The SNMP, blackbox and IPMI exporters are the main examples of this. The IPMI and SNMP exporters as the devices are often black boxes that it’s impossible to run code on (though if you could run a node exporter on them instead that’d be
+better), and the blackbox exporter where you’re monitoring something like a DNS name, where there’s also nothing to run on. In this case, Prometheus should still do service discovery, and pass on the target to be scraped. See the blackbox and SNMP exporters for examples.
 
 Note that it is only currently possible to write this type of exporter with the
 Go, Python and Java client libraries.
 
-The other is where you’re pulling some stats out of a random instance of a
-system and don’t care which one you’re talking to. Consider a set of MySQL
-slaves you wanted to run some business queries against the data to then export.
-Having an exporter that uses your usual load balancing approach to talk to one
-slave is the sanest approach.
+The second exception is where you’re pulling some stats out of a random instance of a system and don’t care which one you’re talking to. Consider a set of MySQL slaves you wanted to run some business queries against the data to then export. Having an exporter that uses your usual load balancing approach to talk to one slave is the sanest approach.
```

Comment:
While we're at it, WDYT about fixing slavery metaphors here? I think "replica" is a perfectly self-explanatory alternative.

Reply:
Awesome - wasn't sure - some projects are receptive to this - others it's a 500 comment storm. Will fix.

Reply:
In this case replica is actually more accurate.